agent/main.go

```diff
@@ -71,6 +72,23 @@ func (l *labelFlags) Set(value string) error {
 	}
 }
 
+func setFlagIfNotSet(flagName string) error {
+	found := false
+	flag.Visit(func(f *flag.Flag) {
+		if f.Name == flagName {
+			found = true
```

Comment:
wouldn't it be optimal to set and break as soon as we find the flagName not set, rathe then iterating all other set flags, we ignoring them anyways!!

Reply:
ignore this, we can't control this as its external to this